### PR TITLE
Added special case support for caddyfile suffix, case insensitive

### PR DESCRIPTION
**Pull request by omalk98**
## Description

- Added case insensitive support for special prefix case `Caddyfile`
- Added support for `.caddyfile` suffix, also case insensitive
- Handles extension check without case sensitivity

References issue [#5909 ](https://github.com/caddyserver/caddy/issues/5909)

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -147,8 +147,10 @@ func loadConfigWithLogger(logger *zap.Logger, configFile, adapterName string) ([
 	// as a special case, if a config file called "Caddyfile" was
 	// specified, and no adapter is specified, assume caddyfile adapter
 	// for convenience
-	if strings.HasPrefix(filepath.Base(configFile), "Caddyfile") &&
-		filepath.Ext(configFile) != ".json" &&
+	baseConfig := strings.ToLower(filepath.Base(configFile))
+	if (strings.HasPrefix(baseConfig, "caddyfile") ||
+		strings.HasSuffix(baseConfig, "caddyfile")) &&
+		filepath.Ext(baseConfig) != ".json" &&
```
Comment (mholt) [inline cmd/main.go:153]: What if they have a `Caddyfile.yaml` etc? :thinking:

File: cmd/main.go [hunk 2]
```diff
@@ -147,8 +147,10 @@ func loadConfigWithLogger(logger *zap.Logger, configFile, adapterName string) ([
 	// as a special case, if a config file called "Caddyfile" was
 	// specified, and no adapter is specified, assume caddyfile adapter
 	// for convenience
-	if strings.HasPrefix(filepath.Base(configFile), "Caddyfile") &&
-		filepath.Ext(configFile) != ".json" &&
+	baseConfig := strings.ToLower(filepath.Base(configFile))
+	if (strings.HasPrefix(baseConfig, "caddyfile") ||
+		strings.HasSuffix(baseConfig, "caddyfile")) &&
```
Comment (mholt) [inline cmd/main.go:152]:
```suggestion
		strings.HasSuffix(baseConfig, ".caddyfile")) &&
```

Probably want to make it a file extension?

**Review by mholt — CHANGES_REQUESTED** [review]
Thanks, sorry for the delay (we had a baby unexpectedly early around that time).

I think I like where this is going but it could use a little refinement. Let me know your thoughts!

File: cmd/main.go (hunk 1 above)
Comment (omalk98) [reply cmd/main.go:153]: Good point, it slipped my mind.
A possible solution is to have a function from the `caddyconfig` package which checks if an extension matches a registered adapter name.

**Comment by omalk98** [comment]
I think this can open up an interesting path for automation. What if the registered modules provide a static list of file extensions in its interface that they are able to handle? Like `yaml` and `yml`, `mysql` and `sql`, etc. 
This could allow bypassing the adapter flag when a known file extension is used. 
What are your thoughts on this? 
I'm willing to open up some pull requests in the adapter modules to support this.

All the best with your new baby!

**Comment by mholt** [comment]
@omalk98 Whew, finally starting to go through my backlog. Anyway, that is interesting; I guess each adapter would just have to add a line of code to register its known file extension(s). That sounds like a good idea though, should help reduce frustrations if forgetting to use the `--adapter` flag. I'd be open to this!

**Review by mholt — APPROVED** [review]

**Comment by mholt** [comment]
@omalk98 I fixed a panic when the config file has no extension (slice access of `[1:0]`) in [e1f4b83](https://github.com/caddyserver/caddy/commit/e1f4b83ffaa861e8f383ba52108e6449b53217f2).

**Pull request merged**